src/sentry/sentry_metrics/consumers/indexer/processing.py

```diff
@@ -33,12 +48,14 @@ def process_messages(
     The value of the message is what we need to parse and then translate
     using the indexer.
     """
-    batch = IndexerBatch(use_case_id, outer_message)
+    batch = IndexerBatch(config.use_case_id, outer_message)
 
     org_strings = batch.extract_strings()
 
+    indexer = STORAGE_TO_INDEXER[config.db_backend]()
```

Comment:
Hold on, you are reinstantiating the indexer object for every message batch you receive. 
That is not needed and actually can be damaging if the indexer instance does something expensive during the instantiation.
Why do you need to make the instantiation of the indexer a Callable ?

Reply:
I was trying to make this work with `CLOUDSPANNER` initialization, we talk a little about it in https://github.com/getsentry/sentry/pull/37745#discussion_r948329920

Reply:
I see the reasoning but it won't be ok. 
If you see how the initialization in Nikhar's PR 
https://github.com/getsentry/sentry/pull/37802/files#diff-6263481e823d6e3fd17a3d6f4a7ffc84b417dd9f1e15db83cc647937536b0399R97-R100
When you initialize the spanner client we establish the TCP connection.
We need to reuse that connection, we cannot re-establish it for every batch..
We generally process tens of batches per second, we cannot establish and destroy the TCP connection each time. Besides the overhead, I would not be surprised if Google throttled us.

So, since the indexer instance is not a static singleton anymore we need an alternative way.
Options:
- Instantiate the indexer in the consumer and pass it to the `process_messages` function.
- Turn the message processing logic into a class so that the indexer can be a local attribute.